### PR TITLE
Myho/beta release

### PR DESCRIPTION
**Pull request by mydiemho**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,12 +17,29 @@ script:
 after_success:
 - cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js && rm -rf ./coverage
 
+before_deploy: 
+  - npm run build
+  - cd lib/
```
Comment (wbreza) [inline .travis.yml]: Why changing directory to lib/ ?
If you are attempting to modify what is deployed as part of the package we can use `files` section of the `package.json`

Comment (wbreza) [reply .travis.yml]: ex) https://github.com/microsoft/vott-react/blob/master/package.json

Comment (mydiemho) [reply .travis.yml]: oh I wasn't aware you could do that, that's pretty cool

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "serverless-azure-functions",
-  "version": "0.7.0",
+  "version": "1.0.0-2",
```
Comment (pjlittle) [inline package.json]: what's the significance of the `-2`, in the semver?

Comment (mydiemho) [reply package.json]: this comes from the prerelease option, `-2` let us bump beta version until we're ready for official release.  

when we're ready, running `npm version major` will turn it into `v1.0.0`

File: .travis.yml [hunk 3]
```diff
@@ -1,28 +1,65 @@
...
+    - stage: publish
+      name: "NPM release"
+      node_js: "10"
+      before_deploy: 
+        - npm run build
+      deploy:
+        - provider: npm
+          email: $NPM_EMAIL_ACCOUNT
+          api_key:
+            secure: Iu5Q2uuiF+jRWkNRtX77YElMwyEopzKD+iHQQkT7HheH12AwDAW/PZiwLRhBTif8kvHmIIwPhjey6i7QwmJFUmpb433299WF+tinRhy5mBFntdaNmgS2Wvv/hOFJza2XdOeIsKA+s+zPAGArfLonmvQX5QKVmJI7/W+hyqJT0FjtTMqxPyJ00ENuDeZdUEcsaoDAwu1XTA6bj5ILnguR+smeGoZscpp8nwTv27fA+GWLFGf08pPAyuHSwc1/lRxSbLlw/f0ZqnCclGU+SF2EV4/9+xxcPWqtJdJfURR/LaOkg3Q5o4aILRjNJNCbpRkrc5lEcPEvBPLrI6e7F9ssg08aHniaMvdD5B9TsgGqV6sxZX/KvMIHWfzBRLuCzY2bPE8Wbw+ZI/yi9+89NZdRikAMyWocvg4tx1OsQRcrRltsXqQrZyw45IR5oSv0kLx54xrslXu8jjP08jQEpnBncJp883DX6DmO42TRP4QYHsQ8BB5hHQ2qmVoKMydkLy0Kdvq7e+9v/pZHEMsR/HOwz1S5xYsSeqwqoq4u/H+hPFWu3GVJXbsVLHZuaDU3fBRMtJOcjujPszrvN/cNt2JryTT1EYPJotmpURqu/n9vPLMzoi1wIKPrZiAcEzh8MLlM93MyG/yuPeHZOvMuwCVtOXd3KKH8sEJvbTVUtKIvQXw=
```
Comment (pjlittle) [inline .travis.yml]: just double checking, but these are the encrypted values from running the travis tool, right? :slightly_smiling_face:

**Comment by pjlittle** [comment]
LGTM :shipit:

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "serverless-azure-functions",
-  "version": "0.7.0",
+  "version": "1.0.0-3",
```
Comment (tbarlow12) [inline package.json]: `-3`?

File: .travis.yml [hunk 5]
```diff
@@ -1,28 +1,55 @@
...
 install:
-- npm install
-script:
-  - npm run test:ci
+  - npm install
 
-after_success:
-- cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js && rm -rf ./coverage
+script:
+  - printenv | sort
```
Comment (pjlittle) [inline .travis.yml]: do you still want this in here?

Comment (mydiemho) [reply .travis.yml]: good catch!

File: scripts/npmPublishBeta.sh [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+# echo "checkout new branch for version bump"
+git checkout -b npmRelease
+
+## configure npm to sign commit
+npm config set sign-git-tag true
+
+version=$(npm version prerelease -m "Bumped to version %s")
+
+git tag -l "v1*"
```
Comment (pjlittle) [inline scripts/npmPublishBeta.sh]: was this for visibility/debugging or does this so something else?

Comment (mydiemho) [reply scripts/npmPublishBeta.sh]: for visibility when running the script.  I'll change it to "echo $version" instead.

File: .travis.yml [hunk 7]
```diff
@@ -1,28 +1,53 @@
...
+      deploy:
+        - provider: npm
+          email: "sls-az@microsoft.com"
+          api_key:
+            secure: hFgX3Ru3/CnT90d5WYXZ6Yr91zsHw+D+ZxaDdimtjMSjhZjKgA6HJ82A9SlpVRIm0WcwVYyBX8nd2j4d+Gkcefd8B4tt/UHOxuzFmt9Q5GvyjdKgA0jzY4XqjUogh142IUJUABRodmh8Wr7mHWCXhNNJu17UUyEf+KRliz72Orp36JEmozdgf+CpzVyzGUL2OLBZSmkjpFTUS4clOVhzwj0G2IHNB7qcupAVQ+r3DOjiU1N4+KkT+yrWH1esqOZYWVcqyVTUgkrfLU1eyaP8IemS+nT3Y9l0VxXY6nnZ4fkENbrivaN5cgp/GPSFb6QSiMbwImaPgxhyj2sIb1ydnoza5lR83b5Tsaa5E90gUbU23kms9J0Qi+6WLaI63kdfvQidl2A5UVpDr+lVusg0gfibkMUo4Jbof67o9coAu+WO80qS0jdn9cAg3gX6Xx9c7fUdHrFfxw4TBxjfI0SO7gJwbZvLZsZ9WsTT/SflIUQL72MhiYTQNIk7ewIVQXjb6eDuZS0tgSu3Gb0W7CIsVW7fwMmMvlEfzkzH119YYVUwwBCACfCgyXaFaamJ1tLHRlp1yjDjkBMZLlbOh18cSrTBu86+RBSX+I1jRvNoUxSTtzZjCI9i7sv9lEP2zq5EEMbA8qjCguu48knqQIrzsPNVCysKrNI964aVb8k4MyY=
+          skip_cleanup: true
+          tag: beta   # tag all npm publish from dev branch with "beta" TODO: remove before merge to master
+          on: # https://docs.travis-ci.com/user/deployment/#conditional-releases-with-on
+            tags: true
+            repo: serverless/serverless-azure-functions
```
Comment (tbarlow12) [inline .travis.yml]: are we allowing tags on any branch to be published? just dev and master?

File: .travis.yml (hunk 3 above)
Comment (tbarlow12) [reply .travis.yml]: Yup 👍

**Pull request merged**